backend/graphql/service/order.service.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { OrderModel } from '../../database/models/order.model'
+import { NotFoundError } from '../errors/notFoundError'
+import {
+  type OrderDeliveredInput,
+  type GetOrderInput,
+  type OrderCanceledInput
+} from '../inputs/order.input'
+import { type Context } from '../types/context.type'
+import { type Order } from '../types/order.type'
+
+export class OrderService {
+  async getAllOrders(): Promise<Order[]> {
+    return await OrderModel.find().populate('user', '-password')
+  }
+
+  async findSingleOrder(input: GetOrderInput): Promise<Order> {
+    const order = await OrderModel.findOne(input)
+
+    if (!order) {
+      throw new NotFoundError('Order not found')
+    }
+
+    return order
+  }
+
+  async updateOrderToDelivered(input: OrderDeliveredInput): Promise<Order> {
+    const order = await OrderModel.findOne({ _id: input._id })
+
+    if (!order) {
+      throw new NotFoundError('Order not found')
+    }
```

Comment:
```suggestion
  const order = await this.findSingleOrder({ _id: input._id })
```

Reply:
Property 'save' does not exist on type 'Order'.

Reply:
What? that makes no sense.. it's literally the same code...